### PR TITLE
Refactoring distributions

### LINKED ISSUE #34: Rename parameters to match TensorFlow Distributions
To make it easy to port models between PyTorch and TensorFlow, it would be great if `torch.distributions` used the same distribution names and parameter names as `tf.contrib.distributions`.

- [x] `Categorical(probs)`
- [x] `Normal(loc, scale)`
- [x] `Dirichlet(concentration)`
- [x] `Beta(concentration1, concentration0)`
- [x] `Gamma(concentration, rate)`
- [x] `Exponential(rate)`
- [ ] What else? See [TF Distributions docs](https://www.tensorflow.org/api_docs/python/tf/distributions)

### PR DESCRIPTION
**Pull request by alicanb**
fixes #34

### CODE REVIEW

File: test/test_distributions.py [hunk 1]
```diff
@@ -1058,24 +1058,24 @@ def test_dirichlet_on_diagonal(self):
...
-            cdf_alpha = (cdf(x, alpha + eps, beta) - cdf(x, alpha - eps, beta)) / (2 * eps)
-            cdf_x = pdf(x, alpha, beta)
+            eps = 0.01 * con1 / (1.0 + np.sqrt(con1))
+            cdf_alpha = (cdf(x, con1 + eps, con0) - cdf(x, con1 - eps, con0)) / (2 * eps)
+            cdf_x = pdf(x, con1, con0)
             expected_grad = -cdf_alpha / cdf_x
             rel_error = np.abs(actual_grad - expected_grad) / (expected_grad + 1e-30)
             self.assertLess(np.max(rel_error), 0.005, '\n'.join([
-                'Bad gradient dx/dalpha for x ~ Beta({}, {})'.format(alpha, beta),
+                'Bad gradient dx/dalpha for x ~ Beta({}, {})'.format(con1, con0),
```
Comment (fritzo) [inline test/test_distributions.py]: Could you update to say `Bad gradient dx/dcon1`?

File: test/test_distributions.py [hunk 2]
```diff
@@ -1088,24 +1088,24 @@ def test_beta_wrt_alpha(self):
...
-            cdf_beta = (cdf(x, alpha, beta + eps) - cdf(x, alpha, beta - eps)) / (2 * eps)
-            cdf_x = pdf(x, alpha, beta)
+            eps = 0.01 * con0 / (1.0 + np.sqrt(con0))
+            cdf_beta = (cdf(x, con1, con0 + eps) - cdf(x, con1, con0 - eps)) / (2 * eps)
+            cdf_x = pdf(x, con1, con0)
             expected_grad = -cdf_beta / cdf_x
             rel_error = np.abs(actual_grad - expected_grad) / (expected_grad + 1e-30)
             self.assertLess(np.max(rel_error), 0.005, '\n'.join([
-                'Bad gradient dx/dbeta for x ~ Beta({}, {})'.format(alpha, beta),
+                'Bad gradient dx/dbeta for x ~ Beta({}, {})'.format(con1, con0),
```
Comment (fritzo) [inline test/test_distributions.py]: Could you update to say `Bad gradient dx/dcon0`?

File: torch/distributions/beta.py [hunk 3]
```diff
@@ -10,36 +10,36 @@
 
 class Beta(Distribution):
     r"""
-    Creates a Beta distribution parameterized by concentration `alpha` and `beta`.
+    Creates a Beta distribution parameterized by concentration `concentration1` and `concentration0`.
```
Comment (fritzo) [inline torch/distributions/beta.py]: nit: Could you wrap docstrings to <=80 characters so that `help(Beta)` works in jupyter notebooks?

**Comment by alicanb** [comment]
Thanks @fritzo , sending upstream
